docs/userguide/glossary.rst

```diff
@@ -0,0 +1,218 @@
+
+.. _nodeglossary:
+
+**Node:**
+------------
+
+A node in Parsl is like a workstation in a factory. It's the physical or virtual machine where work gets done. Each node provides the computational power needed to run tasks, and it can host several workers who carry out the tasks.
+
+.. _parallelismglossary:
+
+**Parallelism:**
+-------------------
+
+Parallelism means doing multiple things at the same time. In Parsl, it means that your apps can run simultaneously on different computers. This can make your programs run much faster.
+
+.. _parslscriptglossary:    
+
+**Parsl Script:**
+---------------------
+
+A Parsl script is a file that contains the instructions for how to run your apps in parallel. It's like a recipe that tells you what ingredients to use and how to combine them.
+
+.. _pluginglossary:
+
+**Plugin:**
+---------------
+
+A plugin is like an add-on for Parsl. It's a piece of code that you can add to Parsl to give it new features or change how it works. It's like an extra tool that you can add to your toolbox.
+
+.. _pythonappglossary: 
+
+**Python App:**
+------------------
+
+A Python app is a special kind of app in Parsl that's written in the Python programming language. It's a way to use Parsl to run your Python code in parallel.
+
+.. _resourceglossary:
+
+**Resource:**
+---------------
+
+A resource in Parsl refers to any computational asset that can be used to execute tasks, such as CPU cores, memory, or entire nodes. It's like the tools and materials you need to get a job done.
+
+.. _serializationglossary:    
+
+**Serialization:**
+--------------------
+
+Serialization is like packing your belongings into a suitcase so you can take them on a trip. In Parsl, it means converting your data into a format that can be sent over a network to another computer.
+
+.. _stagingglossary:    
+
+**Staging:**
+---------------
+
+Staging is like setting the stage for a play. In Parsl, it means preparing the data that your apps need before they start running. This can involve things like copying files to the right location or converting them into the right format.
+
+.. _taskglossary:
+
+**Task:**
+------------
+
+A task in Parsl is the smallest unit of work that can be executed. It's like a single step in a larger process, where each task is part of a broader workflow or job.
+
+.. _threadglossary:    
+
+**Thread:**
+-------------
+
+A thread is like a smaller part of a program that can run independently. It's like a worker in a factory who can do their job at the same time as other workers.
+
+.. _workerglossary:
+
+**Worker:**
+-------------
+
+A worker in Parsl is like an employee in the factory who does the actual work. Workers run on nodes and are responsible for executing the tasks assigned to them. Multiple workers can work on a single node, sharing the node’s resources to get the job done efficiently.
```

Comment:
Worker and Thread seem similar in definition here, but I don't think they should be.